R/paste.R

```diff
@@ -3,78 +3,214 @@
 #' `dm_paste()` takes an existing `dm` and emits the code necessary for its creation.
 #'
 #' @inheritParams dm_add_pk
-#' @param select Boolean, default `FALSE`. If `TRUE` will produce code
-#'   for reducing to necessary columns.
+#' @param select
+#'   Deprecated, see `"select"` in the `options` argument.
+#' @param ... Must be empty.
 #' @param tab_width Indentation width for code from the second line onwards
+#' @param options Formatting options. A character vector containing some of:
+#'   - `"tables"`: [tibble()] calls for empty table definitions
+#'     derived from [dm_ptype()], overrides `"select"`.
+#'   - `"select"`: [dm_select()] statements for columns that are part
+#'     of the dm.
+#'   - `"pks"`: [dm_add_pk()] statements for primary keys.
+#'   - `"keys"`: [dm_add_fk()] statements for foreign keys, implies `"pks"`.
```

Comment:
Why not just have one option for setting keys? Do you have a use case when FKS should be explicitly avoided?
```suggestion
#'   - `"keys"`: [dm_add_pk()] and [dm_add_fk()] statements for adding keys.
```
If this suggestion is accepted, further changes later on in the code are required